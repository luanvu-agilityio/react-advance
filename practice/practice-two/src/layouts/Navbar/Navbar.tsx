import { ChevronDown, ChevronRight, Menu, X } from 'lucide-react'
import { useNavigate } from 'react-router-dom'
import { useCallback, useState } from 'react'
import { navbarData } from '@data/navbar'
import {
  HoverContent,
  HoverMenuItem,
  HoverSubContent,
  HoverSubTrigger,
  MenubarRoot,
  MenuTitle,
  MenuWrapper,
  MobileMenu,
  MobileMenuButton,
  MobileMenuHeader,
  MobileSubItem,
  NavbarWrapper,
  StyledAccordionContent,
  StyledAccordionItem,
  StyledAccordionRoot,
  StyledAccordionTrigger,
  StyledTriggerButton,
} from './Navbar.styles'

// Helper function to convert label to URL path
const labelToPath = (label: string): string => {
  return label.toLowerCase().replace(/\s+/g, '-')
}

export const Navbar = () => {
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false)
  const navigate = useNavigate()

  const handleCategoryClick = useCallback(
    (label: string) => {
      const path = labelToPath(label)
      navigate(`/${path}`)
    },
    [navigate]
  )

  const handleSubItemClick = useCallback(
    (href: string) => {
      navigate(href)
    },
    [navigate]
  )

  return (
    <NavbarWrapper className="section">
      <MobileMenuButton onClick={() => setIsMobileMenuOpen(true)}>
        <Menu size={24} style={{ marginLeft: '16px' }} />
      </MobileMenuButton>

      {/* Desktop Menu with Hover */}
      <MenubarRoot>
        {navbarData.map((item, index) => (
          <MenuWrapper key={`menu-${item.label}-${index}`}>
            <StyledTriggerButton
              onClick={() => handleCategoryClick(item.label)}
            >
              {item.label}{' '}
              <ChevronDown
                size={12}
                style={{
                  color: 'var(--green-color-default)',
                  strokeWidth: 4,
                }}
              />
            </StyledTriggerButton>

            <HoverContent className="dropdown-content">
              {item.categories?.map((category, catIndex) => (
                <HoverSubTrigger key={`category-${category.title}-${catIndex}`}>
                  <span>{category.title}</span>
                  <ChevronRight
                    size={12}
                    style={{
                      color: 'var(--green-color-default)',
                      strokeWidth: 4,
                    }}
                  />

                  <HoverSubContent className="sub-dropdown">
                    {category.items.map((subItem, itemIndex) => (
                      <HoverMenuItem
                        key={`item-${category.title}-${subItem.title}-${itemIndex}`}
                        onClick={() => handleSubItemClick(subItem.href)}
                      >
                        {subItem.title}
                      </HoverMenuItem>
                    ))}
                  </HoverSubContent>
                </HoverSubTrigger>
              ))}
            </HoverContent>
          </MenuWrapper>
        ))}
      </MenubarRoot>

      {/* Mobile Menu - Keep existing implementation */}
      <MobileMenu isOpen={isMobileMenuOpen}>
        <MobileMenuHeader>
          <MenuTitle>Menu</MenuTitle>
          <MobileMenuButton onClick={() => setIsMobileMenuOpen(false)}>
            <X size={24} />
          </MobileMenuButton>
        </MobileMenuHeader>

        <StyledAccordionRoot type="multiple">
          {navbarData.map((item, index) => (
            <StyledAccordionItem
              key={`mobile-menu-${item.label}-${index}`}
              value={item.label}
            >
              <StyledAccordionTrigger>
                <span>{item.label}</span>
                <ChevronDown
                  size={16}
                  style={{
                    color: 'var(--green-color-default)',
                    strokeWidth: 4,
                  }}
                />
              </StyledAccordionTrigger>
              <StyledAccordionContent>
                {item.categories?.map((category, catIndex) => (
                  <StyledAccordionItem
                    key={`mobile-category-${category.title}-${catIndex}`}
                    value={`${item.label}-${category.title}`}
                  >
                    <StyledAccordionTrigger>
                      <span>{category.title}</span>
                      <ChevronDown
                        size={16}
                        style={{
                          color: 'var(--green-color-default)',
                          strokeWidth: 4,
                        }}
                      />
                    </StyledAccordionTrigger>
                    <StyledAccordionContent>
                      {category.items.map((subItem, itemIndex) => (
                        <MobileSubItem
                          key={`mobile-item-${subItem.title}-${itemIndex}`}
                          onClick={() => {
                            navigate(subItem.href)
                            setIsMobileMenuOpen(false)
                          }}
                        >
                          {subItem.title}
                        </MobileSubItem>
                      ))}
                    </StyledAccordionContent>
                  </StyledAccordionItem>
                ))}
              </StyledAccordionContent>
            </StyledAccordionItem>
          ))}
        </StyledAccordionRoot>
      </MobileMenu>
    </NavbarWrapper>
  )
}
