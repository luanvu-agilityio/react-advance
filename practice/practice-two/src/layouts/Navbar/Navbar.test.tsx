import { TextEncoder } from 'util'
global.TextEncoder = TextEncoder

import { fireEvent, render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Navbar } from './Navbar'

// SNAPSHOT TESTING
describe('Navbar - Rendering', () => {
  test('matches desktop snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })

  test('renders mobile menu button', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )

    const mobileButton = container.querySelector('button')
    expect(mobileButton).toBeInTheDocument()
  })

  test('renders desktop menu with categories', () => {
    const { getAllByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )

    expect(getAllByText(/Bakery/i)[0]).toBeInTheDocument()

    expect(getAllByText(/Bakery/i).length).toBeGreaterThan(0)
  })
})

// INTERACTIVE BEHAVIOR TEST

const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}))

describe('Navbar - Interactive', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('navigates when category is clicked in desktop view', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )

    // Find the first category button in desktop menu
    const firstCategory = container.querySelector(
      '[data-testid="category-button"]'
    )

    if (firstCategory) {
      fireEvent.click(firstCategory)

      // Check that navigate was called with the correct path
      // The exact path depends on your navbarData
      expect(mockNavigate).toHaveBeenCalled()
      expect(mockNavigate.mock.calls[0][0]).toMatch(/^\//)
    } else {
      throw new Error('No category buttons found in desktop menu')
    }
  })

  test('opens and closes mobile menu', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )

    // Find mobile menu button
    const mobileMenuButton = container.querySelector('button')

    if (!mobileMenuButton) {
      throw new Error('Mobile menu button not found')
    }

    // Click to open mobile menu
    fireEvent.click(mobileMenuButton)

    // Check menu is open by looking for the title
    expect(screen.getByText('Menu')).toBeInTheDocument()

    // Find close button (using different methods to be more reliable)
    const closeButton =
      screen.getByRole('button', { name: /close/i }) ||
      screen
        .getAllByRole('button')
        .find(
          (btn) =>
            btn.getAttribute('aria-label') === 'Close menu' ||
            btn.textContent?.includes('Close')
        ) ||
      container.querySelector('[data-testid="close-button"]')

    if (!closeButton) {
      throw new Error('Close button not found in mobile menu')
    }

    // Click to close mobile menu
    fireEvent.click(closeButton)

    // Verify the menu is closed (the Menu title should no longer be visible)
    expect(screen.queryByText('Menu')?.closest('[data-state]')).toHaveAttribute(
      'data-state',
      'closed'
    )
  })
})
