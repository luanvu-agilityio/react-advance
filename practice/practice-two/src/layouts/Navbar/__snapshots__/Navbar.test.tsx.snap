// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar - Rendering matches desktop snapshot 1`] = `
<div>
  <div
    class="sc-Qotzb cUSTmH section"
  >
    <button
      class="sc-dKREkF iYShG"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-menu sc-dFaThA kZjXbD"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 12h16"
        />
        <path
          d="M4 18h16"
        />
        <path
          d="M4 6h16"
        />
      </svg>
    </button>
    <div
      class="sc-fYsHOw dIRERo"
      data-orientation="horizontal"
      role="menubar"
      style="outline: none;"
      tabindex="-1"
    >
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Fruit and vegetables
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Fresh Fruits
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Apples
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Bananas
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Oranges
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Strawberries
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Dragon Fruit
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Vegetables
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Tomatoes
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Baby Spinach
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Organic Carrots
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Organic
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Organic Avocados
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Organic Berry Mix
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Bakery
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Bread
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Sourdough Bread
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                French Baguette
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Pastries
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Butter Croissants
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Cakes
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Birthday Cake
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Meat and fish
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Beef
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Ribeye Steak
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Ground Beef
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Poultry
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Chicken Breasts
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Fish
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Wild Caught Salmon
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Fresh Jumbo Shrimp
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Drinks
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Non-Alcoholic
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Orange Juice
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Green Tea
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Alcoholic
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Craft IPA
              </div>
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Organic Red Wine
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Special Drinks
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Berry Smoothie
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Kitchen
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Cookware
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Frying Pan Set
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Appliances
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Stand Mixer
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Accessories
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Chef Knife Set
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Special nutrition
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Supplements
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Protein Powder
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Diets
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Keto Shakes
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Superfoods
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Chia Seeds
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Baby
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Food
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Baby Formula
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Care
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Eco-Friendly Diapers
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Accessories
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Feeding Set
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jJAtPt dqqxLN"
      >
        <button
          class="sc-cZSric juElHV"
          data-testid="category-button"
        >
          Pharmacy
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down sc-lixPIL ecxPpi"
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div
          class="sc-cUiCeM dZEcRt dropdown-content"
        >
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Medications
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Cold & Flu Relief
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Personal Care
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                Advanced Skincare
              </div>
            </div>
          </div>
          <div
            class="sc-jPkiSJ dwTfyH"
          >
            <span>
              Medical Supplies
            </span>
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right sc-etfXYe dGDdFJ"
              fill="none"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <div
              class="sc-jScdur cYaggx sub-dropdown"
            >
              <div
                class="sc-dcKlJK bjrQfS"
              >
                First Aid Kit
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
